corehq/apps/api/tests/core_api.py

```diff
@@ -539,11 +549,40 @@ def test_search_param(self):
             "/a/test_domain/api/v0.5/form/",
             data={'_search': json.dumps(query)}
```

Comment:
The `query` submitted here uses ES2 syntax. I'm assuming the API logic does something to transform it to ES7 syntax. Does this mean our API will support ES2 query syntax indefinitely?

Reply:
No we don't plan to support the query at all. This sort of calls are coming from Data Export Tool, and one custom use-case, only these two we are supporting. Earlier, I had worked on [this PR](https://github.com/dimagi/commcare-hq/pull/28032) that discontinues support for random JSON ES queries from ES.  From that PR (which has been merged a while ago)
> Remove the ability to pass query part of elasticsearch query in public APIs. I made sure that this is not being used on prod by doing greps for these (grep 'api/v' | grep 'q=' and then grep 'api/v' | grep 'query=' over all proxy logs)
> Only accept two type of ES queries from _search GET param from known use cases. These I found based on grepping prod for _search param on api urls.